utils/frame/benchmarking-cli/src/writer.rs

```diff
@@ -30,12 +30,15 @@ pub fn open_file(path: &str) -> Result<File, std::io::Error> {
 		.open(path)
 }
 
-pub fn write_trait(file: &mut File, batches: Result<Vec<BenchmarkBatch>, String>) -> Result<(), std::io::Error> {
-	let batches = batches.unwrap();
-
+pub fn write_trait(file: &mut File, batches: Vec<BenchmarkBatch>) -> Result<(), std::io::Error> {
 	let mut current_pallet = Vec::<u8>::new();
 
+	// Skip writing if there are no batches
+	if batches.is_empty() { return Ok(()) }
+
 	batches.iter().for_each(|batch| {
+		// Skip writing if there are no results
+		if batch.results.is_empty() { return }
```

Comment:
Shouldn't that be `continue`?

Reply:
`for_each` takes a closure, so I am returning early on that closure. Should be the same effect, but `continue` would give a compile error.

Reply:
You are 100% correct. I did not realize that you in a closure. I think `for batch in batches {}`would be more idiomatic, though. Then you would need `continue`.

Reply:
sure! its all the same to me. I actually thought `iter().for_each()` was the way all the cool rust people do things. Is there any general guidance here?

Reply:
Could not find anything. I just read `for ... in` more and also prefer it myself for obvious reasons: Better readability. You save the closure syntax and one set of braces.

I think it comes down to taste.

Reply:
done!